travel-event-host-app/src/app/api/auth/[...nextauth]/route.ts

```diff
@@ -4,18 +4,59 @@ import { connectMongoDB } from '@/lib/mongodb';
 import User from '@/schemas/user';
 import { compare } from 'bcrypt';
 
+type SessionUser = {
+  name?: string | null | undefined;
+  _id: string;
+  firstName?: string;
+  imageUrl?: string;
+  lastName?: string;
+  email: string;
+  location: {
+    country: string;
+    state: string;
+    city: string;
+    coords: {
+      lat: number;
+      long: number;
+    };
+  };
+  eventIds: string[];
+  admin?: boolean;
+};
```

Comment:
It would be helpful to write a comment on why this type is declared here and what is a `SessionUser` in terms of NextAuth. 

It looks similar to the `User` interface (model). Can we not `extend` the `User` interface to be used here? If we change or rename properties of the User model, we have to remember to do it here too.  
```suggestion
interface SessionUser extends User {
   ...// add properties that are in addition to User
};
```

Reply:
this type is for the callbacks, the User have sensitive data, also the User is a interface this have to be a type mandatory

Reply:
if you a way to extend from the user but remove properties like password for example and without changing it to interface